art_tree.go

```diff
@@ -17,40 +20,90 @@ func NewArtTree() *ArtTree {
 	return &ArtTree{root: nil, size: 0}
 }
 
+type Result struct {
+	Key   string
+	Value interface{}
+}
+
+func (t *ArtTree) EachChanResult() chan Result {
+	return t.EachChanResultFrom(t.root)
+}
+
+func (t *ArtTree) EachChanResultFrom(start *ArtNode) chan Result {
+	outChan := make(chan Result)
+	go func() {
+		if start != nil {
+			for n := range t.EachChanFrom(start) {
+				if n.IsLeaf() {
+					outChan <- Result{string(n.key), n.value}
+				}
+			}
+		}
+		close(outChan)
+	}()
+	return outChan
+}
+
+// Finds the starting node for a prefix search and returns an array of all the objects under it
+func (t *ArtTree) PrefixSearch(key []byte) []interface{} {
+	ret := make([]interface{}, 0)
+	for r := range t.PrefixSearchChan(key) {
+		ret = append(ret, r.Value)
+	}
+	return ret
+}
```

Comment:
This is an interesting approach.

It's not formally specified in the research paper I used to base this data structure on, but I think it could be a neat way of being able to get a subtree of values, should others want a specific prefix.

Curious, is there a reason why you chose to append all of the leaf nodes together instead of just returning a subtree?